client/src/Components/FilterSection/index.tsx

```diff
@@ -2,51 +2,120 @@ import React, { useContext, useState, useEffect } from 'react';
   const locationData = useLocation();
-  const search : any = locationData.state;
-  const { serviceSearch, locationSearch } = search;
-
   const [mainServiceId, setMainServiceId] = useState(0);
   const [subService, setSubService] = useState<number[]>([]);
   const [error, setError] = useState('');
   const [subServices, setSubServices] = useState<serviceObject[]>([]);
-
+  const [name, setName] = useState<string>('');
+  const [locationInput, setLocationInput] = useState<number>();
+  const search : any = locationData.state;
+  const { serviceSearch, locationSearch } = search;
   const { data: { location, services } } = useContext(ServiceLocation);
   const { Option } = Select;
-
-  useEffect(() => {
-    const getSubServicesData = async () => {
-      try {
-        setError('');
-        const { data } : any = await request('get', `/subservices/${+mainServiceId || serviceSearch}`);
-        setSubServices(data);
-      } catch (responseError: any) {
-        setError(responseError.data.msg);
-      }
-    };
-    if (mainServiceId || serviceSearch) getSubServicesData();
-  }, [mainServiceId, serviceSearch]);
-
+  const getSearchResults = async (queryString: string) => {
+    try {
+      setSearchError('');
+      const searchResult : SearchResponse = await request('get', `/provider?${queryString}`);
+      setIsLoading(false);
+      setResultCount(searchResult.count);
+      setUsers(searchResult.data);
+    } catch (responseError: any) {
+      setSearchError(responseError.data.msg);
+    }
+  };
+  const getSubServicesData = async () => {
+    try {
+      setError('');
+      const { data } : any = await request('get', `/subservices/${+mainServiceId || serviceSearch}`);
+      setSubServices(data);
+    } catch (responseError: any) {
+      setError(responseError.data.msg);
+    }
+  };
   const handelSelectMainService = async (service: number) => {
     setSubService([]);
     setMainServiceId(service);
+    setPage(1);
+  };
+  const handelNameInputChange = async (e: Event) => {
+    setName(e.target.value);
+    setPage(1);
+  };
+  const handelLocationInput = async (e: number) => {
+    setLocationInput(e);
+    setPage(1);
   };
+  useEffect(() => {
+    let queryString = '';
+    if (serviceSearch) {
+      queryString += `&service=${serviceSearch}`;
+    }
+    if (locationSearch) {
+      queryString += `&location=${locationSearch}`;
+    }
+    if (page !== 1) {
+      queryString += `&page=${page}`;
+    }
+    if (queryString) {
+      getSearchResults(queryString);
+    }
+  }, []);
+  useEffect(() => {
+    let queryString = '';
+    if (mainServiceId) {
+      queryString += `&service=${mainServiceId}`;
+    }
+    if (subService.length) {
+      queryString += `&subservice=${subService.join(',')}`;
+    }
+    if (name) {
+      queryString += `&name=${name}`;
+    }
+    if (locationInput) {
+      queryString += `&location=${locationInput}`;
+    }
+    if (page !== 1) {
+      queryString += `&page=${page}`;
```

Comment:
No need to this condition
Just in the end before sending the request add 
```suggestion
      queryString += `&page=${page}`;
```